crates/miden-objects/src/errors.rs

```diff
@@ -429,6 +448,90 @@ pub enum ProvenTransactionError {
     },
 }
 
+// BATCH ERROR
+// ================================================================================================
+
+#[derive(Debug, Error)]
+pub enum BatchProposeError {
```

Comment:
Don't have a strong opinion here, but why did we change `BatchError` to `BatchProposeError`? If `BatchError` was too generic, should this be `ProposedBatchError` instead?